src/webgpu/api/operation/render_pass/resolve.spec.ts

```diff
@@ -0,0 +1,206 @@
+  [0, 2],
+  [1, 3],
+  [0, 1, 2, 3],
+];
+
+const kStoreOps: GPUStoreOp[] = ['clear', 'store'];
+
+const kSize = 4;
+const kFormat: GPUTextureFormat = 'rgba8unorm';
+export const g = makeTestGroup(GPUTest);
+
+g.test('render_pass_resolve')
+  .params(
+    params()
+      .combine(poptions('numColorAttachments', kNumColorAttachments))
+      .combine(poptions('slotsToResolve', kSlotsToResolve))
+      .combine(poptions('storeOperation', kStoreOps))
+      .combine(poptions('resolveTargetBaseMipLevel', kBaseMipLevel))
+      .combine(poptions('resolveTargetBaseArrayLayer', kBaseArrayLayers))
+  )
+  .fn(t => {
+    // These shaders will draw a white triangle into a texture. After draw, the top left
+    // half of the texture will be white, and the bottom right half will be unchanged. When this
+    // texture is resolved, there will be two distinct colors in each portion of the texture, as
+    // well as a line between the portions that contain the midpoint color due to the multisample
+    // resolve.
+    const vertexModule = t.makeShaderModule('vertex', {
+      glsl: `
+      #version 450
+      const vec2 pos[3] = vec2[3](vec2(-1.0f, -1.0f), vec2(-1.0f, 1.0), vec2(1.0, 1.0));
+        void main() {
+          gl_Position = vec4(pos[gl_VertexIndex], 0.0f, 1.0f);
+        }
+      `,
+    });
+
+    const fragmentModule = t.makeShaderModule('fragment', {
+      glsl: `
+      #version 450
+        layout(location = 0) out vec4 fragColor0;
+        layout(location = 1) out vec4 fragColor1;
+        layout(location = 2) out vec4 fragColor2;
+        layout(location = 3) out vec4 fragColor3;
+        void main() {
+          fragColor0 = vec4(1.0, 1.0, 1.0, 1.0);
+          fragColor1 = vec4(1.0, 1.0, 1.0, 1.0);
+          fragColor2 = vec4(1.0, 1.0, 1.0, 1.0);
+          fragColor3 = vec4(1.0, 1.0, 1.0, 1.0);
+        }
+      `,
+    });
+
+    const colorStateDescriptors: GPUColorStateDescriptor[] = [];
+    for (let i = 0; i < t.params.numColorAttachments; i++) {
+      colorStateDescriptors.push({ format: kFormat });
+    }
+
+    const pipeline = t.device.createRenderPipeline({
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      primitiveTopology: 'triangle-list',
+      rasterizationState: {
+        frontFace: 'ccw',
+      },
+      colorStates: colorStateDescriptors,
+      vertexState: {
+        indexFormat: 'uint32',
+        vertexBuffers: [],
+      },
+      sampleCount: 4,
+    });
+
+    const resolveTargets: GPUTexture[] = [];
+    const renderPassColorAttachmentDescriptors: GPURenderPassColorAttachmentDescriptor[] = [];
+
+    // The resolve target must be the same size as the color attachment. If we're resolving to mip
+    // level 1, the resolve target base mip level should be 2x the color attachment size.
+    const kResolveTargetSize: number = t.params.resolveTargetBaseMipLevel === 1 ? kSize * 2 : kSize;
```

Comment:
This can be computed more simply as:
```suggestion
    const kResolveTargetSize = kSize << t.params.resolveTargetBaseMipLevel;
```

Reply:
Done.